core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/EnvelopeConcat.java

```diff
@@ -47,21 +49,37 @@ private static List<LocatedEnvelope> initLocatedEnvelopes(List<? extends Envelop
     }
 
     @Override
-    public double interpolateTotalTime(double position) {
+    public double interpolateArrivalAt(double position) {
+        var envelope = findEnvelopeAt(position);
+        assert envelope != null : "Trying to interpolate time outside of the envelope";
+        return envelope.startTime + envelope.envelope.interpolateArrivalAtClamp(position - envelope.startOffset);
+    }
```

Comment:
We *may* still have an ambiguity there.

`findEnvelopeAt` doesn't have clear semantics on which part is returned on transitions, and I believe it's not even consistent. If there is a stop right at a transition we need to pick the right side.

I believe we should change it to `findEnvelopeLeft` / right, and get the left version for arrival and right for departure.

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/7609/commits/b7f044ebd08664f02948f7aafb19982fee469f65